src/utils/serve-functions.js

```diff
@@ -417,32 +425,43 @@ const getBuildFunction = ({ functionBuilder, log }) =>
     }
   }
 
-const setupFunctionsBuilder = async ({ site, log, warn }) => {
-  const functionBuilder = await detectFunctionsBuilder(site.root)
-  if (functionBuilder) {
-    log(
-      `${NETLIFYDEVLOG} Function builder ${chalk.yellow(
-        functionBuilder.builderName,
-      )} detected: Running npm script ${chalk.yellow(functionBuilder.npmScript)}`,
-    )
-    warn(
-      `${NETLIFYDEVWARN} This is a beta feature, please give us feedback on how to improve at https://github.com/netlify/cli/`,
-    )
+const setupFunctionsBuilder = async ({ config, errorExit, functionsDirectory, log, site, warn }) => {
+  const functionBuilder = await detectFunctionsBuilder({
+    config,
+    errorExit,
+    functionsDirectory,
+    log,
+    projectRoot: site.root,
+  })
 
-    const debouncedBuild = debounce(getBuildFunction({ functionBuilder, log }), 300, {
-      leading: true,
-      trailing: true,
-    })
+  if (!functionBuilder) {
+    return {}
+  }
 
-    await debouncedBuild()
+  const npmScriptString = functionBuilder.npmScript
+    ? `: Running npm script ${chalk.yellow(functionBuilder.npmScript)}`
+    : ''
 
-    const functionWatcher = chokidar.watch(functionBuilder.src)
-    functionWatcher.on('ready', () => {
-      functionWatcher.on('add', debouncedBuild)
-      functionWatcher.on('change', debouncedBuild)
-      functionWatcher.on('unlink', debouncedBuild)
-    })
-  }
+  log(`${NETLIFYDEVLOG} Function builder ${chalk.yellow(functionBuilder.builderName)} detected${npmScriptString}.`)
+  warn(
+    `${NETLIFYDEVWARN} This is a beta feature, please give us feedback on how to improve at https://github.com/netlify/cli/`,
```

Comment:
@erezrokah Do we still want to display this message for people using the ZISI function builder?

Reply:
Great catch - I forgot about it. Let's remove it all together

Reply:
Done in https://github.com/netlify/cli/pull/2094/commits/4ad71a4960407fc2397ac485c07acb331245b490.